clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/isolated/IsolatedIngestionServerHandler.java

```diff
@@ -201,6 +201,7 @@ private IngestionTaskReport handleIngestionTaskCommand(IngestionTaskCommand inge
           isolatedIngestionServer.cleanupTopicState(topicName);
           break;
         case REMOVE_PARTITION:
+          isolatedIngestionServer.maybeSubscribeNewResource(topicName, partitionId);
```

Comment:
Can you explain why need to call `subscribe` when removing a partition?

Reply:
It is caused by the check mechanism implemented here. When the server started, nothing has been initialized. When II process received a command, it should check whether the TP has been "initialized" (subscribed), if it is either not subscribed or has been ingested and hand-over to main (unsubscribed), II process should reject the command, as it is unexpected. To adapt to the change in this PR, we need to let the II process accept the remove partition command if the TopicPartition has never been processed by II. This will be the same behavior as the OFFLINE -> STANDBY.
The command will only change the TP's subscription state to true if it is never found in the state map. When a TP is handed over to main process, it will be changed to false.

Reply:
I can also try to add some explanation in the PR

Reply:
The current logic sound weird to me..
So What is the logic when II is disabled to handle such case, does Venice Server need to try to subscribe? I don't recall such logic.

Reply:
All this subscribe thingy is for II only.
We build the subscription metadata for each topic partition in main process and isolated process, this is needed as it can direct any helix command to go to the place where the resource is hosted. This is for correctness guarantee. 
For a new resource that metadata management does not recognize, by default the OFFLINE->STANDBY goes to II. For this PR's purpose, it change to make OFFLINE->DROPPED by default goes to II as well.

For non II and one server process only, all the command goes to the same process so we don't need the metadata management at all.

Reply:
It just looks weird to put this logic here.
Can the following `dropStoragePartitionGracefully` function ignore whether the to-be-dropped partition has subscribed before or not?
If we put it here, I think it will be confusing to other people even through it can run correctly.
People might get illusion that before dropping a particular partition, Venice Server needs to subscribe it first and subscription in Server normally means: create the partition, subscribe to Kafka and start ingestion and more.
I recommend either ignore the check in the following function call or rename this function to something else since `subscribe` is abused here from my understanding.

Reply:
I see your concern, let me think about it...